File: test/acceptance/pages/AllGroupsPage.java
package acceptance.pages;

import acceptance.PlayContainer;
import acceptance.SharedDriver;
import org.junit.Assert;
import org.openqa.selenium.By;
import org.openqa.selenium.NoSuchElementException;

public class AllGroupsPage {
  private final SharedDriver driver;

  public AllGroupsPage(SharedDriver driver) {
    this.driver = driver;
  }

  public boolean isViewing() {
    return "Grupper".equals(driver.findElement(By.id("page_name")).getText());
  }

  public boolean verifyGroupListed(String groupName) {
    try {
      driver.findElement(By.xpath("//a[contains(., '" + groupName + "')]"));
      return true;
    } catch (NoSuchElementException ex) {
      return false;
    }
  }

  public AllGroupsPage navigateTo() {
    driver.navigate().to(PlayContainer.getBaseUrl() + "/groups");
    Assert.assertTrue("Not on groups page!", isViewing());
    return this;
  }

  public AllGroupsPage selectGroup(String groupName) {
    driver.findElement(By.xpath("//a[contains(., '" + groupName + "')]")).click();
    return this;
  }
}

File: test/se/crisp/signup4/acceptance/pages/AllGroupsPage.scala
package se.crisp.signup4.acceptance.pages

import org.openqa.selenium.{By, WebDriver}

class AllGroupsPage(baseUrl: String, driver: WebDriver) extends Page(baseUrl, driver) {

  def isViewing: Boolean = "Grupper" == driver.findElement(By.id("page_name")).getText

  def verifyGroupListed(groupName: String): Boolean = try {
    driver.findElement(By.xpath("//a[contains(., '" + groupName + "')]"))
    true
  } catch {
    case _: NoSuchElementException =>
      false
  }

  def navigateTo() {
    driver.navigate.to(baseUrl + "/groups")
    waitForCompletePageLoad()
  }

  def selectGroup(groupName: String) {
    driver.findElement(By.xpath("//a[contains(., '" + groupName + "')]")).click()
    waitForCompletePageLoad()
  }

}
